tests/test_raw_node.rs

```diff
@@ -141,6 +141,56 @@ fn test_raw_node_propose_and_conf_change() {
     assert_eq!(entries[1].get_data(), &*ccdata);
 }
 
+// test_raw_node_read_index ensures that RawNode.read_index sends the MsgReadIndex message
+// to the underlying raft. It also ensures that ReadState can be read out.
+#[test]
+fn test_raw_node_read_index() {
```

Comment:
The function can be wrapped into an option. Actually this is the original implementation of step function pointer. I replaced it with a bool and assert for simplicity because all the related cases just panicked in the function hook.